os/arch/arm/src/amebad/amebad_i2s.c

```diff
@@ -0,0 +1,1818 @@
+			goto errout;
+		}
+	}
+#endif
+#if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
+	if (priv->txenab) {
+		priv->tx.dog = wd_create();
+		if (!priv->tx.dog) {
+			i2serr("ERROR: Failed to create the TX DMA watchdog\n");
+			goto errout;
+		}
+	}
+#endif
+	return OK;
+
+errout:
+#if defined(I2S_HAVE_RX) && (0 < I2S_HAVE_RX)
+	if (priv->rx.dog){
+		wd_delete(priv->rx.dog);
+		priv->rx.dog = NULL;
+	}
+#endif
+#if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
+	if (priv->tx.dog){
+		wd_delete(priv->tx.dog);
+		priv->tx.dog = NULL;
+	}
+#endif
+	return -ENOMEM;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name:
+ *
+ * Description:
+ *   Initialize the selected i2s port
+ *
+ * Input Parameter:
+ *   none
+ *
+ * Returned Value:
+ *   Valid i2s device structure reference on succcess; a NULL on failure
+ *
+ ****************************************************************************/
+struct i2s_dev_s *amebad_i2s_initialize(uint16_t port)
+{
+	if (port >= I2S_NUM_MAX) {
+		i2serr("ERROR: Port number outside the allowed port number range\n");
+		return NULL;
+	}
+
+	if (g_i2sdevice[port] != NULL) {
+		return &g_i2sdevice[port]->dev;
+	}
+
+	struct amebad_i2s_s *priv;
+	int ret;
+
+	/* Allocate a new state structure for this chip select.  NOTE that there
+	 * is no protection if the same chip select is used in two different
+	 * chip select structures.
+	 */
+	priv = (struct amebad_i2s_s *)kmm_zalloc(sizeof(struct amebad_i2s_s));
+	if (!priv) {
+		i2serr("ERROR: Failed to allocate a chip select structure\n");
+		return NULL;
+	}
+
+	/* Get default configuration */
+	i2s_getdefaultconfig(priv);
+
+	/* Config values initialization */
+	amebad_i2s_initpins(priv);
+
+	/* Initialize buffering */
+#if defined(I2S_HAVE_RX) && (0 < I2S_HAVE_RX)
+	i2s_buf_rx_initialize(priv);
+	priv->rxenab = 1;
+#endif
+#if defined(I2S_HAVE_TX) && (0 < I2S_HAVE_TX)
+	i2s_buf_tx_initialize(priv);
+	priv->txenab = 1;
+#endif
+
+	/* I2s object initialization */
+	i2s_init(&priv->i2s_object, priv->i2s_sclk_pin, priv->i2s_ws_pin, priv->i2s_sd_tx_pin, priv->i2s_sd_rx_pin, priv->i2s_mck_pin);
+
+	/* Initialize buffering */
+	//i2s_set_dma_buffer(&priv->i2s_object, priv->i2s_tx_buf, priv->i2s_rx_buf, I2S_DMA_PAGE_NUM, I2S_DMA_PAGE_SIZE);
+
+	/* configures IRQ */
+	priv->rx_isr_handler = (i2s_irq_handler)&i2s_transfer_rx_handleirq;
+	priv->tx_isr_handler = (i2s_irq_handler)&i2s_transfer_tx_handleirq;
+	ret = amebad_i2s_isr_initialize(priv);
+	if (ret != OK) {
+		i2serr("I2S initialize: isr fails\n");
+		goto errout_with_alloc;
+	}
+
+	/* Initialize the I2S priv device structure  */
+	sem_init(&priv->exclsem, 0, 1);
+	priv->dev.ops = &g_i2sops;
+
+	ret = i2s_allocate_wd(priv);
+	if (ret != OK) {
+		i2serr("I2S initialize: watch dog alloc fails\n");
```

Comment:
I think we don't need to print this log, because in `i2s_allocate_wd()`, we already print the log.

Reply:
Removed the above mentioned log.